sbt-dotty/src/dotty/tools/sbtplugin/DottyPlugin.scala

```diff
@@ -169,6 +170,12 @@ object DottyPlugin extends AutoPlugin {
   override def projectSettings: Seq[Setting[_]] = {
     Seq(
       isDotty := scalaVersion.value.startsWith("0.") || scalaVersion.value.startsWith("3."),
+      isDottyJS := {
```

Comment:
The way this works basically assumes that the settings of `ScalaJSPlugin` will be applied *before* those of `DottyPlugin`. It seems to be the case in the tests I did, perhaps because `ScalaJSPlugin` is explicitly enabled, while `DottyPlugin` is triggered. However, I could not find an authoritative source on the topic.

I have an alternative implementation in mind, that would not have that assumption: it would be to have *another* `DottyJSPlugin`, that would be auto-triggered by the presence of `DottyPlugin` *and* `ScalaJSPlugin`. That plugin would be guaranteed to have its settings be applied after both of them, by the documented rules. However, that would require `sbt-dotty` to depend on `sbt-scalajs` to be able to refer to `ScalaJSPlugin`.

Reply:
Hopefully in the not-too-distant future sbt-dotty will be obsoleted by sbt supporting dotty natively, so we need to plan for that too

Reply:
When that happens, the Scala.js-Dotty combination will have to be handled by sbt-scalajs. Currently though, since apparently `DottyPlugin` gets applied after `ScalaJSPlugin`, there's nothing sbt-scalajs can do, AFAICT.

Reply:
> When that happens, the Scala.js-Dotty combination will have to be handled by sbt-scalajs.

OK, can you add a comment mentioning this around here?

Reply:
Done.